src/ApiGateways/Web.Bff.Shopping/aggregator/Controllers/BasketController.cs

```diff
@@ -1,164 +1,154 @@
-﻿using Microsoft.AspNetCore.Authorization;
-using Microsoft.AspNetCore.Mvc;
-using Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Models;
-using Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Services;
-using System;
-using System.Linq;
-using System.Net;
-using System.Threading.Tasks;
-
-namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Controllers
+﻿namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Controllers;
+
+[Route("api/v1/[controller]")]
+[Authorize]
+[ApiController]
+public class BasketController : ControllerBase
 {
-    [Route("api/v1/[controller]")]
-    [Authorize]
-    [ApiController]
-    public class BasketController : ControllerBase
+    private readonly ICatalogService _catalog;
+    private readonly IBasketService _basket;
+
+    public BasketController(ICatalogService catalogService, IBasketService basketService)
     {
-        private readonly ICatalogService _catalog;
-        private readonly IBasketService _basket;
+        _catalog = catalogService;
+        _basket = basketService;
+    }
 
-        public BasketController(ICatalogService catalogService, IBasketService basketService)
+    [HttpPost]
+    [HttpPut]
+    [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+    [ProducesResponseType(typeof(BasketData), (int)HttpStatusCode.OK)]
+    public async Task<ActionResult<BasketData>> UpdateAllBasketAsync([FromBody] UpdateBasketRequest data)
+    {
+        if (data.Items == null || !data.Items.Any())
         {
-            _catalog = catalogService;
-            _basket = basketService;
+            return BadRequest("Need to pass at least one basket line");
         }
 
-        [HttpPost]
-        [HttpPut]
-        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
-        [ProducesResponseType(typeof(BasketData), (int)HttpStatusCode.OK)]
-        public async Task<ActionResult<BasketData>> UpdateAllBasketAsync([FromBody] UpdateBasketRequest data)
+        // Retrieve the current basket
+        var basket = await _basket.GetById(data.BuyerId) ?? new BasketData(data.BuyerId);
```

Comment:
If the previous API change is made that I requested, this would need to change too.

```suggestion
        var basket = await _basket.GetByIdAsync(data.BuyerId) ?? new BasketData(data.BuyerId);
```

Reply:
Yes, correct.